api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -195,6 +195,10 @@ model ApiProfile {
   /** The internet visibility of the OpenShift API server */
   @visibility(Lifecycle.Read, Lifecycle.Create)
   visibility?: Visibility = "public";
+
+  /** The IP ranges authorized to access the Kubernetes API server. */
+  @visibility(Lifecycle.Read, Lifecycle.Create, Lifecycle.Update)
+  authorizedIpRanges?: string[];
```

Comment:
Given that this is to the kubernetes api server, should the attribute name be more specific? For example, would it limit access in the ingress, or other endpoints?

Reply:
My read of the [XCMSTRAT](https://issues.redhat.com/browse/XCMSTRAT-517) is this is only for the restricting API server access.

IMO, its placement in `ApiProfile` along with the doc comment makes the context clear enough.
```
{
    "properties": {
        "api" {
            "url": "https://...",
            "visibility": "Public",
            "authorizedIpRanges": [
                "192.168.1.0/24"
            ]
        }
    }
}
```

If we wanted something similar for ingress I imagine we'd add an `IngressProfile` model.